### ISSUE TITLE
[ENH] Let `fmt: skip` apply to compound statements, if they share a line.

### ISSUE THREAD

**Issue by randolf-scholz** [astral-sh/ruff#11216]
```python
if True: x = 0  # fmt: skip

def foo(self, lorem, ipsum, dolor, sit, amet, consectetur, adipiscing, elit, sed, do): ...  # fmt: skip
```

Gets reformatted to

```python
if True:
    x = 0  # fmt: skip

def foo(
    self, lorem, ipsum, dolor, sit, amet, consectetur, adipiscing, elit, sed, do
): ...  # fmt: skip
```

Which is because `fmt: skip` only applies to the preceding `statement`. However, it would be useful to apply the `fmt: skip` to the whole [compound statement](https://docs.python.org/3/reference/compound_stmts.html) in this context. For instance, when formatting `@overload` decorated functions.

**Comment by MichaReiser** [comment]
Yeah that makes sense to me. I'm not entirely sure how to implement this yet. I see two approaches:

1. Add special handling for `fmt: skip` and `fmt: off` (and I think `yapf:disable`) comments in `placement.rs` that come at the end of a clause header. Instead of associating them with the preceding statement, associate them with the clause header (but only if both are on the same line). 
2. Pass the comments of the first simple statement to `FormatClauseHeader` and make the decision there. 

The good news, the unused formatter suppression comment lint rule seems to already accept this.

**Comment by bnkc** [comment]
Hi @MichaReiser I did a first pass on this issue. Not sure if my approach is what we were looking for but I'd be happy to change direction. Let me know 👍

**Comment by AlexanderHott** [comment]
Any update on this issue/PR?
